sql/src/main/java/io/druid/sql/calcite/expression/RowExtraction.java

```diff
@@ -124,7 +124,11 @@ public boolean isFilterable(final RowSignature rowSignature)
     return Filters.FILTERABLE_TYPES.contains(rowSignature.getColumnType(column));
   }
 
-  public DimensionSpec toDimensionSpec(final RowSignature rowSignature, final String outputName)
+  public DimensionSpec toDimensionSpec(
+      final RowSignature rowSignature,
+      final String outputName,
+      final ValueType outputType
```

Comment:
This should have a `Preconditions.checkNotNull(outputType, "outputType")` to prevent bugs due to accidentally forgetting to pass in an outputType. Callers should pass in STRING if that's what they want.